src/osd/OSD.h

```diff
@@ -613,7 +615,7 @@ class OSDService : public Scrub::ScrubSchedListener {
 private:
   // -- pg recovery and associated throttling --
   ceph::mutex recovery_lock = ceph::make_mutex("OSDService::recovery_lock");
-  std::list<std::pair<epoch_t, PGRef> > awaiting_throttle;
+  std::list<std::tuple<epoch_t, PGRef, int> > awaiting_throttle;
```

Comment:
See my branch, let's use a named struct here.

Reply:
Sure, accessing the tuple members did seem awkward. This way it's clear which member is being accessed.